### DISCUSSION TITLE
[Help request]: Help with understanding of hyprland layout

### DISCUSSION THREAD

**Discussion by kekulta** [JaKooLit/Hyprland-Dots#217] [General]
### Hyprland layout

Hi! I've been using your dots(installed with Fedora-Hyprland script) for a few weeks and it's been pretty good expirience.
But now I wanted a little bit of my own touch in the system, so I reinstalled it and started to recreate your config, I wanted to achieve the same behavior but with my dots structure/themes/better understanding/etc
With some trouble I recreate almost everything I needed but now I tried to set up windows layout and couldn't achieve the same behavior as at system with your config.

When I used your config it was some mix of dwindle and master layouta when one window is always took the left half of the screen and others form a tree at the right half of the screen. When a new window created it tooks left side of the screen and the previous "master" pushed to the tree at the left side. When new window closed previous layout restored and the old "master" again tooks it place.

But now I can have either master layout with one "master" and a stack of "slaves" on top of each other(with broken navigation too! I can't use super+arrow, it just won't work in the "slaves" stack) or dwindle, but new windows won't took the "master" position and it's fairly hard to have everything in order.

Initially, I thought that I need to use these settings from your configs:

```
dwindle {
  pseudotile = yes
  preserve_split = yes
  special_scale_factor = 0.8
}

master {
  new_is_master=1
  new_on_top=1
  mfact = 0.5
}

general {
  sensitivity=1.00
  apply_sens_to_raw=1
  gaps_in = 4
  gaps_out = 8
  border_size = 2
  resize_on_border = true
   
  col.active_border = $color0 $color2 $color4 $color6 $color8 90deg
  col.inactive_border = $backgroundCol

  layout = master
}
```

But when I used it nothing really changed. I read all of the documentation for layouts on the hyprland wiki but it looks like what I got is an expected behavior. I tried copy your window rules(and read documentation about them as well) but still got nothing. Generally, it feels like what I got now is how things should work. I did not find any mentions about this combined master-dwindle layout.

Is it expected behavior? Did this was some sort of a bug that merged this two layouts at my system or you somehow intentionally achieved that behavior? If yes then how did you get it and how can I reproduce it?

I'm starting feeling now that all these few weeks were my fever dreams and I just used master layout. But I really don't want to reinstall my system once again to check if that actually happened.

**Comment by JaKooLit** [comment]
layout = master

you can just simple change this to

layout = dwindle

and if you want to switch between dwindle to master, just press super alt L then it will change layout accordingly

but after reboot, or logout / login, it will set to which ever layout you set

**Comment by kekulta** [comment]
Yes, that what I tried and I did not get the behavior I got with your config installed.
Here are some screenshots.
For first and second screenshots I just spammed new terminals in master and dwindle layout.
For third one I manually created window order I tried to achieve(and the one I got when I installed your config!). Its like a spiral dwindle layout where there is one "master" window(the one with echo "Sixth window"). So if I opened one more(Seventh window) it will take place of the Sixth window. Sixth window will take place of the fifth window. Fifth window goes to fourth, fourth to third etc.
![20240402_08h17m06s_grim](https://github.com/JaKooLit/Hyprland-Dots/assets/33986203/baf89087-2efc-4722-944b-b4e2b51d38df)
![20240402_08h18m09s_grim](https://github.com/JaKooLit/Hyprland-Dots/assets/33986203/64344c9c-c699-4979-aa8f-e80a229c0be8)
![20240402_08h20m53s_grim](https://github.com/JaKooLit/Hyprland-Dots/assets/33986203/08bf3dec-3b8d-4c15-93ce-917ee726b838)

**Comment by daxisunder** [comment]
The first screenshot is the `master` layout and the third one is the  `dwindle` layout. I don't know about the second one.

**Comment by JaKooLit** [comment]
> The first screenshot is the `master` layout and the third one is the `dwindle` layout. I don't know about the second one.



same as dwindle

**Comment by JaKooLit** [comment]
https://wiki.hyprland.org/Configuring/Master-Layout/

https://wiki.hyprland.org/Configuring/Dwindle-Layout/


hyprland has a good wiki

**Comment by daxisunder** [comment]
> > The first screenshot is the `master` layout and the third one is the `dwindle` layout. I don't know about the second one.
> 
> same as dwindle

What do you mean "same as `dwindke`"? No, it's not.

**Comment by daxisunder** [comment]
> [wiki.hyprland.org/Configuring/Master-Layout](https://wiki.hyprland.org/Configuring/Master-Layout/)
> 
> [wiki.hyprland.org/Configuring/Dwindle-Layout](https://wiki.hyprland.org/Configuring/Dwindle-Layout/)
> 
> hyprland has a good wiki

I know.

**Comment by JaKooLit** [comment]
> > > The first screenshot is the `master` layout and the third one is the `dwindle` layout. I don't know about the second one.
> > 
> > 
> > same as dwindle
> 
> What do you mean "same as `dwindke`"? No, it's not.

zzzzzz


dwindle, will create another window on focused window. But you can change that behaviour on your settings

![Screenshot_22-Apr_13-50-12_23164](https://github.com/JaKooLit/Hyprland-Dots/assets/85185940/3a260419-5740-47a1-9450-8efc3a455e51)

**Comment by daxisunder** [comment]
> > > > The first screenshot is the `master` layout and the third one is the `dwindle` layout. I don't know about the second one.
> > > 
> > > 
> > > same as dwindle
> > 
> > 
> > What do you mean "same as `dwindke`"? No, it's not.
> 
> zzzzzz
> 
>> dwindle, will create another window on focused window. But you can change that behaviour
>
>>Of course you can change it, but it's not the default behavior.
